clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -848,7 +849,8 @@ else if (enableAutoCommit)
                   long requestTimeoutMs,
                   int defaultApiTimeoutMs,
                   List<ConsumerPartitionAssignor> assignors,
-                  String groupId) {
+                  String groupId,
```

Comment:
this changes the constructor signature. for compatibility please add a constructor with the old signature that calls this one supplying a default value for skipMetadataCacheUpdateUponUnassignment (which should be false for safety)